### PR TITLE
Update Dependencies and test on Elixir 1.14 and 1.15

### PR DESCRIPTION
**Pull request by kyleboe**

### CODE REVIEW

**Comment by kyleboe** [comment]
I opened up a [PR](https://github.com/kyleboe/wallaby/pull/1) on my fork so I could run CI and it seems to be failing for the same reason as the Selenium v4 macOS pipeline. It is pulling a newer version of elixir and there are deps that are incompatible with elixir 1.14+

**Comment by mhanberg** [comment]
@kyleboe the v4 jobs aren't expected to succeed. I just have them in there in case someone wants to look into #569

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,11 @@ jobs:
 
     strategy:
       matrix:
-        elixir-otp: [{otp: 24.x, elixir: 1.12.x}, {otp: 25.x, elixir: 1.13.x}]
+        elixir-otp:
+          - otp: 25.x
+            elixir: 1.14.x
+          - otp: 26.x
+            elixir: 1.15.x
```
Comment (mhanberg) [inline .github/workflows/ci.yml:17]: why did you remove 1.12 and 1.13?

File: test/wallaby/http_client_test.exs [hunk 2]
```diff
@@ -73,7 +73,7 @@ defmodule Wallaby.HTTPClientTest do
 
     test "includes the original HTTPoison error when there is one", %{bypass: bypass} do
       expected_message =
-        "Wallaby had an internal issue with HTTPoison:\n%HTTPoison.Error{id: nil, reason: :econnrefused}"
+        "Wallaby had an internal issue with HTTPoison:\n%HTTPoison.Error{reason: :econnrefused, id: nil}"
```
Comment (mhanberg) [inline test/wallaby/http_client_test.exs:76]: if this is flaky, i would do multiple `=~` asserts instead

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kyleboe) [reply .github/workflows/ci.yml:17]: Because they are not backwards compatible with the changed dependencies.

Comment (kyleboe) [reply .github/workflows/ci.yml:17]: This will likely require a major version bump to signify this change.

File: test/wallaby/http_client_test.exs (hunk 2 above)
Comment (kyleboe) [reply test/wallaby/http_client_test.exs:76]: I'm not confident that it is flaky just yet. I want to understand why this change was necessary.

Comment (mhanberg) [reply test/wallaby/http_client_test.exs:76]: otp changed how maps work for keys < 32 so now key order is not deterministic like it used to be

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mhanberg) [reply .github/workflows/ci.yml:17]: I don't want to drop support for older versions of Elixir just for a dev dependency that I have never even used since taking over the project.

you can probably add `--only test` to mix deps.get in CI to avoid having to update benchee.

File: test/wallaby/http_client_test.exs (hunk 2 above)
Comment (kyleboe) [reply test/wallaby/http_client_test.exs:76]: Agreed. But if maps < 32 keys are treated like keyword lists in memory, they are linked and therefore have a deterministic order yes? So my inference would be that `{:id, nil}` was added before `{:reason, :econnrefused}` in the internals of HTTPoison, thereby putting `:reason` at the head of the linked list.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kyleboe) [reply .github/workflows/ci.yml:17]: I am down to drop benchee as a dependency altogether. However there are other changes around `ExUnit.Server.modules_loaded()` that are not backwards compatible.

File: test/wallaby/http_client_test.exs (hunk 2 above)
Comment (mhanberg) [reply test/wallaby/http_client_test.exs:76]: I'm trying to say that in OTP26 that is no longer the case. the order is nondeterministic

https://www.erlang.org/blog/otp-26-highlights/#changed-ordering-of-atom-keys

You can pass `custom_options: [sort_maps: true]` to as an Inspect.Opt tho, which will sort it and it should be deterministic.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mhanberg) [reply .github/workflows/ci.yml:17]: that is a private elixir API, and is in a unit test.

I would just add a conditional that checks if the function is exported and call the one that is

`function_exported?(ExUnit.Server, :modules_loaded, 0)`, etc

File: test/wallaby/http_client_test.exs (hunk 2 above)
Comment (kyleboe) [reply test/wallaby/http_client_test.exs:76]: Ah. I see! Thanks for explaining 😃 I'll make some more adjustments and see if I can get things backwards compatible as well. Thanks for the feedback!

Comment (mhanberg) [reply test/wallaby/http_client_test.exs:76]: Of course 😁
